### DISCUSSION TITLE
dnSpy - ILSpy

### DISCUSSION THREAD

**Discussion by NN---** [dnSpyEx/dnSpy#226] [Q&A]
How much is possible to contribute dnSpy features to ILSpy ?
This could improve both products.

**Comment by ElektroKill** [comment]
dnSpy is licensed under GPL v3 and ILSpy is licensed under MIT. Code from dnSpy cannot be taken verbatim and placed in a different project under a different license. dnSpy also shares very little code with ILSpy. ILSpy and dnSpy are also designed and maintained with different use cases in mind. ILSpy is primarily a decompiler and dnSpy is more of an all-in-one .NET reverse engineering suite. While dnSpy does use the ILSpy decompilation engine, it is a heavily modified version of a really old ILSpy version (2.4) to allow integration with other dnSpy systems as well as better behavior when dealing with heavily obfuscated assemblies. Work is being done to move on to the latest decompiler engine.

Merging the two projects is pretty much impossible.
